### ISSUE TITLE
How to load my own trained model for inference.

### ISSUE THREAD

**Issue by ariel415el** [Tencent-Hunyuan/Hunyuan3D-2.1#59]
Hey, I'm using your mini dataset exampole and I struggle with loading the trained model
The problem is that the config files and the ckpts does not have the same structure as the pretrained ones and the code expects the later.
For example the two yamls below are different

1. https://huggingface.co/tencent/Hunyuan3D-2.1/blob/main/hunyuan3d-dit-v2-1/config.yaml
2. https://github.com/Tencent-Hunyuan/Hunyuan3D-2.1/blob/main/hy3dshape/configs/hunyuan3ddit-mini-overfitting-flowmatching-dinog518-bf16-lr1e4-512.yaml

I've trained the model using
`python3 main.py --config configs/hunyuandit-mini-overfitting-flowmatching-dinog518-bf16-lr1e4-512.yaml --output_dir output_folder/dit/overfitting
`

I'm loading the model with 
```
root_path = '/common_data/workspace/ariel/repos/Hunyuan3D-2.1'
my_ckpt = f'{root_path}/hy3dshape/output_folder/dit/overfitting/ckpt/ckpt-step=00084000.ckpt'
# config_path = f'{root_path}/hy3dshape/configs/hunyuandit-mini-overfitting-flowmatching-dinog518-bf16-lr1e4-512.yaml'
config_path = f'{root_path}/hy3dshape/configs/hunyuan3ddit-mini-overfitting-flowmatching-dinog518-bf16-lr1e4-512.yaml'
shape_pipeline = Hunyuan3DDiTFlowMatchingPipeline.from_single_file(
    ckpt_path=my_ckpt,
    config_path=config_path,
    device='cuda',
    dtype=torch.float16,
)
mesh_untextured = shape_pipeline(image='image.jpg')[0]
mesh_untextured.export('output.obj')
```

I'm using diffusers=='0.30.0' if it matters

Is this the correct way to do inference with my model?

**Comment by ariel415el** [comment]
As a workaround I'm  using this lines in **Hunyuan3DDiTPipeline.from_single_file**

`
  model = instantiate_from_config(config['model']['params']['denoiser_cfg'])
  vae = instantiate_from_config(config['model']['params']['first_stage_config'])
  image_processor = instantiate_from_config(config['model']['params']['image_processor_cfg'])
  conditioner = instantiate_from_config(config['model']['params']['cond_stage_config'])
  scheduler_config = {
      "target": "hy3dshape.schedulers.FlowMatchEulerDiscreteScheduler",
      "params": {
          "num_train_timesteps": 1000
      }
  }
  scheduler = instantiate_from_config(scheduler_config)
  
  for sub_module, key in [(vae, 'first_stage_model'),
                   (conditioner, 'cond_stage_model'),
                   (model, 'model')]:
      sub_state_dict = {k.replace(f"{key}.", ""): v for k,v in ckpt['state_dict'].items() if k.startswith(key)}
      sub_module.load_state_dict(sub_state_dict, strict=True)
`
